compliance/collection/utils/verify_duration.go

```diff
@@ -0,0 +1,20 @@
+package utils
+
+import (
+	"time"
+
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// VerifyAndUpdateDuration ensures that a given duration is positive bigger than zero and returns a default otherwise
+func VerifyAndUpdateDuration(duration time.Duration) time.Duration {
+	if (duration) <= 0 {
+		log.Warn("Negative or zero duration found. Setting to 4 hours.")
+		return time.Hour * 4
```

Comment:
`time.Hour * 4` is very specific for node rescan interval, therefore this function cannot be presented as generic "util". I think you should simply define it as `GetNodeRescanInterval` in `pkg/env/node_scan.go`. Then, instead of doing this

```go
rescanInterval := complianceUtils.VerifyAndUpdateDuration(env.NodeRescanInterval.DurationSetting())
```

you can simply do this

```go
rescanInterval := env.GetNodeRescanInterval()
```

Reply:
I don't think we need to worry about this because `DurationSetting()` already handles this:

```
// DurationSetting returns the Duration object represented by the environment variable
func (d *DurationSetting) DurationSetting() time.Duration {
	val := os.Getenv(d.envVar)
	if val != "" {
		dur, err := time.ParseDuration(val)
		if err == nil {
			return dur
		}
		log.Warnf("%s is not a valid environment variable for %s, using default value: %s", val, d.envVar, d.defaultDuration.String())
	}
	return d.defaultDuration
}
```

That being said, in another PR, we should probably add a `panic` in the `registerDurationSetting` if the default is not valid

Reply:
https://github.com/stackrox/stackrox/pull/3558

Reply:
ah actually `time.ParseDuration` does allow for negative durations, so `DurationSetting()` does not handle this. Perhaps we should include that in the PR above ^

Reply:
Thanks for linking the PR! We'll keep an eye on it and update our code as soon as your PR is merged.
For now, we want to get this PR merged rather quick, though.